.gitignore

```diff
@@ -50,3 +50,13 @@ src/arm-none-eabi
 
 # clangd
 .cache/
+
+# Toolchain and External dependencies
+sdk-toolchain/
+
+# Developer files
+CMakeUserPresets.json
+compile_commands.json
```

Comment:
Is this needed? From what I understand compile_commands always lives in the `build` folder which is already gitignore'd